docs/usage/validation_errors.md

```diff
@@ -3,251 +3,1470 @@ with some suggestions on how to fix them.
+a tuple, list or a dictionary.
 
 ## `assertion_error`
 
+This error is raised when there is a failing assertion during the validation.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel, field_validator
+
+
+class Model(BaseModel):
+    x: int
+
+    @field_validator('x')
+    @classmethod
+    def force_x_positive(cls, v):
+        assert v > 0
+        return v
+
+Model(x=-1)
+```
 
 ## `bool`
 
 
 ## `bool_parsing`
 
+This error is raised when the value is an invalid boolean string.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    x: bool
+
+print(Model(x='test'))
+```
 
 ## `bool_type`
 
+This error is raised when the value type is not valid for a boolean field.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    x: bool
+
+Model(x=None)
+```
 
 ## `bytes_too_long`
 
+This error is raised when the bytes value length is greater than `Field.max_length`.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel, Field
+
+
+class Model(BaseModel):
+    x: bytes = Field(max_length=3)
+
+Model(x=b'test')
+```
 
 ## `bytes_too_short`
 
+This error is raised when the bytes value length is less than `Field.min_length`.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel, Field
+
+
+class Model(BaseModel):
+    x: bytes = Field(min_length=3)
+
+Model(x=b't')
+```
 
 ## `bytes_type`
 
+This error is raised when the value type is not valid for a bytes field.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    x: bytes
+
+Model(x=123)
+```
 
 ## `callable_type`
 
+This error is raised when the value is not a `callable`.
+
+```py test="skip" lint="skip" upgrade="skip"
+from typing import Any, Callable
+
+from pydantic import BaseModel, ImportString
+
+
+class Model(BaseModel):
+    x: ImportString[Callable[[Any], Any]]
+
+
+Model(x='os.path')
+```
+
+Valid example:
+
+```py test="skip" lint="skip" upgrade="skip"
+from typing import Any, Callable
+
+from pydantic import BaseModel, ImportString
+
+
+class Model(BaseModel):
+    x: ImportString[Callable[[Any], Any]]
+
+
+Model(x='math.cos')
+```
 
 ## `dataclass_type`
 
+This error is raised when the value is not valid for a dataclass field.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import dataclasses
+
+
+@dataclasses.dataclass
+class Nested:
+    x: int
+
+
+@dataclasses.dataclass
+class Model:
+    y: Nested
+
+
+Model(y=1)
+```
+
+Valid example:
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import dataclasses
+
+
+@dataclasses.dataclass
+class Nested:
+    x: int
+
+
+@dataclasses.dataclass
+class Model:
+    y: Nested
+
+
+Model(y=Nested(x=1))
+```
 
 ## `date_from_datetime_inexact`
 
+This error is raised when the datetime value provided for a date field is not an exact date.
```

Comment:
```suggestion
This error is raised when the datetime value provided for a date field has a nonzero time component.
```

I'm not sure if my above comment is exactly right, but I think to a newcomer it may not necessarily be clear what is meant by "an exact date". In particular, if someone is opening this link, they were probably hoping for more clarity than that phrase provides, I think we can explain a bit more clearly here. (Something like "For a timestamp to parse into a field of type `date`, the time components must be all zero, and show an example that doesn't fail below.)